src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -154,43 +72,21 @@ const CustomFieldInputSingleChoice = forwardRef(function CustomFieldInputSingleC
 
   return (
     <div ref={wrapperRef} className={props.className}>
-      <AbstractCustomField
-        icon={caretIcon}
-        clear={clearIcon()}
+      <Select
+        displayValueEvaluator={selectValue => selectValue.label}
+        errorText={props.errorText}
         id={props.id}
         label={props.label}
+        listOptionRefs={listOptionRefs}
         name={props.name}
-        onChange={onSearchChange}
-        onClick={onClick}
-        onKeyDown={onKeyDown}
+        onChange={selectOnChangeHandler}
         placeholder={props.placeholder}
         readOnly={props.readOnly}
-        inputRef={inputRef}
         required={props.required}
-        errorText={validationMessage}
-        value={searchValue || defaultValue}
-        inputRole={'combobox'}
-        ariaProps={{
-          autocomplete: 'list',
-          controls: `${props.id}-single-choice-listbox`,
-          expanded: showOptions,
-          haspopup: 'listbox',
-        }}
-      />
-      { showOptions && (
-        choices.length === 0 ? (<NoOptions className={styles['no-options']} />) : (
-          <Listbox
-            className={styles.dropdown}
-            id={`${props.id}-single-choice-listbox`}
-            labelledBy={`${props.id}-label`}
-            onChange={onSelectionChange}
-            refs={refs}
-            value={value}
-          >
-            { listOptions(choices) }
-          </Listbox>
-        )
-      )}
+        value={value}
+      >
+        { listOptions(choices) }
```

Comment:
Can we inline the choices generation or simplify the chain of functions into just 1 function call? i think we have a chain of functions that only really ever get called once -- an artifact of probably the previous large single choice component.

Reply:
Can do!